resolver/resolver.cc

```diff
@@ -852,8 +852,16 @@ class ResolveTypeParamsWalk {
             core::LambdaParam *parentType = nullptr;
             auto parentMember = data->owner.data(ctx)->superClass().data(ctx)->findMember(ctx, data->name);
             if (parentMember.exists()) {
-                parentType = core::cast_type<core::LambdaParam>(parentMember.data(ctx)->resultType.get());
-                ENFORCE(parentType != nullptr);
+                if (parentMember.data(ctx)->isTypeMember()) {
+                    parentType = core::cast_type<core::LambdaParam>(parentMember.data(ctx)->resultType.get());
+                    ENFORCE(parentType != nullptr);
+                } else {
+                    if (auto e = ctx.state.beginError(send->loc, core::errors::Resolver::InvalidTypeMemberBounds)) {
```

Comment:
Maybe `ParentTypeBoundsMismatch` instead?

Reply:
ack, wasn't sure which error to use (probably not worth creating a new error).